lib/core/test/unit/Cardano/Wallet/DB/SqliteSpec.hs

```diff
@@ -471,6 +484,88 @@ fileModeSpec =  do
+
+                let mockApply h mockTxs = do
+                        Just cpA <- atomically $ readCheckpoint testPk
+                        let slotA = slotId $ currentTip cpA
+                        let Quantity bhA = blockHeight $ currentTip cpA
+                        let hashA = headerHash $ currentTip cpA
+                        let fakeBlock = Block
+                                (BlockHeader
+                                { slotId = slotA { epochNumber = (epochNumber slotA) + 1}
+                                -- Increment blockHeight by steps greater than k
+                                -- Such that old checkpoints are always pruned.
+                                , blockHeight = Quantity $ bhA + 5000
+                                , headerHash = h
+                                , parentHeaderHash = hashA
+                                })
+                                mockTxs
+                                []
+                        let (FilteredBlock _ txs, cpB) = applyBlock fakeBlock cpA
+                        atomically $ do
+                            unsafeRunExceptT $ putCheckpoint testPk cpB
+                            unsafeRunExceptT $ putTxHistory testPk txs
+                            unsafeRunExceptT $ prune testPk
+
+                let mockApplyBlock1 = mockApply (dummyHash "block1")
+                            [ Tx (dummyHash "tx1")
+                                [(TxIn (dummyHash "faucet") 0, Coin 4)]
+                                [ TxOut (head ourAddrs) (Coin 4)
+                                ]
+                            ]
+
+                -- Slot 1 0
+                mockApplyBlock1
+                getAvailableBalance db `shouldReturn` 4
+
+
+                -- Slot 2 0
+                mockApply (dummyHash "block2a")
+                            [ Tx
+                                (dummyHash "tx2a")
+                                [ (TxIn (dummyHash "tx1") 0, Coin 4)
+                                ]
+                                [ TxOut (dummyAddr "faucetAddr2") (Coin 2)
+                                , TxOut (ourAddrs !! 1) (Coin 2)
+                                ]
+                            ]
+
+                -- Slot 3 0
+                mockApply (dummyHash "block3a") []
+                getAvailableBalance db `shouldReturn` 2
+                getTxsInLedger db `shouldReturn` [(Outgoing, 2), (Incoming, 4)]
+
+                -- Since we increment the blockHeight faster than naturally,
+                -- The wallet should have pruned everything but the first and last
+                -- checkpoint.
+                --
+                -- So when we rollback to the previous step...
+                atomically . void . unsafeRunExceptT $
+                    rollbackTo testPk (SlotId 2 0)
+                Just cp <- atomically $ readCheckpoint testPk
+
+                -- We end up at genesis.
```

Comment:
That comment is superfluous. Comments should give detail about "why" , not "what". What is written just below and already perfectly readable :ok_hand:

Reply:
> That comment is superfluous

Yes, perhaps. I totally agree in general.

My reasoning might have been: There is a lot of boilerplate here and "What"-comments highlight the important stuff.

Reply:
Functions should be used then, they are clearer than comments, and re-usable.

Reply:
755065b